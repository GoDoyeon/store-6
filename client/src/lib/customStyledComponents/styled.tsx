import constructWithTag from '.';
import domElements from './utils/domElements';

interface Styled {
  [tag: string]: (string: TemplateStringsArray, ...args: unknown[]) => Function;
}

const styled: Styled = {};

domElements.forEach((domElement) => {
  styled[domElement] = constructWithTag(domElement);
});

export default styled;
