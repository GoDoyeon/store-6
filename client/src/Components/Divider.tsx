import React, { FC } from "react";
import styled from "styled-components";
import underlineSVG from '~/assets/underline.svg';

const DividerFull = styled.div`
  content: "";
  width: 100%;
  padding-left: 2rem;
  height: 1rem;
  background: url(${underlineSVG}) bottom left no-repeat;
  box-sizing: border-box;
  background-size: contain;
`;

const Divider80 = styled.div`
  content: "";
  width: 80%;
  padding-left: 2rem;
  height: 1rem;
  background: url(${underlineSVG}) bottom left no-repeat;
  box-sizing: border-box;
  background-size: contain;
`;

const Divider: FC<{ width: "full" |  "80%" }> = ({ width }) => {
  if (width === "full") return <DividerFull />
  if (width === "80%") return <Divider80 />
};

export default Divider;
