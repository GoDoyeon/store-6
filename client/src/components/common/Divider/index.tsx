import styled from 'styled-components';
import { LineSVG } from '~/assets';

/*
 * @example
 * <Divider width="70%" />
 * <Divider width="32px" />
 * <Divider thick direction="horizontal" />
 */
const Divider = styled.div<{
  width?: string;
  thick?: boolean;
  direction?: 'vertical' | 'horizontal';
}>`
  width: ${({ width }) => width || '100%'};
  height: ${({ thick }) => (thick ? '4px' : '1px')};
  transform: ${({ direction }) =>
    direction === 'vertical' ? 'rotate(90deg)' : 'none'};
  background: url(${LineSVG}) bottom left no-repeat;
  background-size: cover;
  position: relative;
  right: 10px;
`;

export default Divider;
