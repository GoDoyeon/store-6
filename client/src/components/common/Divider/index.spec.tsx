import React from 'react';
import styled from 'styled-components';
import { render } from '@testing-library/react';
import 'jest-styled-components';

import Divider from '~/components/common/Divider';

describe('<Divider />', () => {
  it('should render same with snapshot', () => {
    const { container } = render(<Divider />);
    expect(container).toMatchSnapshot();
  });
});
