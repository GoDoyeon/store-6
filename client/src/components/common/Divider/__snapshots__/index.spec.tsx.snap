// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Divider /> should render same with snapshot 1`] = `
.c0 {
  width: 100%;
  height: 1px;
  -webkit-transform: none;
  -ms-transform: none;
  transform: none;
  background: url(https://store-6-bucket.s3.ap-northeast-2.amazonaws.com/common/line.svg) bottom left no-repeat;
  background-size: cover;
}

<div>
  <div
    class="c0"
  />
</div>
`;
